### PR TITLE
add unit test uuid function

### PR DESCRIPTION
**Pull request by cazterk**
- [ ] unit test check 

working test

### CODE REVIEW

File: itez/beneficiary/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from django.test import TestCase
+import uuid
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: built in imports should come first before django imports

File: itez/beneficiary/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class UuidTest(TestCase):
+    def test_uuid_function(self):
+        uuid_one = generate_uuid_and_agent_code()[0]
+        agent_code = generate_uuid_and_agent_code()[1]
+
+        self.assertEquals(str(uuid_one)[1:8].upper(), str(
+            agent_code).upper().len())
+
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]:
```
class UUIDAndAgentCode(TestCase):
    def SetUp(self)
       self.uuid_code = ...
       self.agent_code = ...

    def TearDown(self):
          pass

    def test_uuid_valid(self):
           assert len(self.uuid_code)  ==  36

    def test_agent_code_length(self):
         assert len(str(self.agent_code)) == 7

    def test_agent_code_upper(self):
          assert self.agent_code == self.agent_code.upper()

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/tests/test_utils.py (hunk 2 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] tests passing

tests

File: itez/beneficiary/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        return super().setUp(self)
+
+    def test_uuid_valid(self):
+        assert len(str(self.uuid_code)) == 36
+
+    def test_agent_code_length(self):
+        assert len(str(self.agent_code)) == 7
+
+    def test_agent_code_upper(self):
+        assert self.agent_code == self.agent_code.upper()
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: Test this with assertEqual
Please read documentation 
Me I just give you ideas not the actual code to be added

File: itez/beneficiary/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    def tearDown(self) -> None:
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: Teardown should be defined just after setup it still runs at the end

File: itez/beneficiary/tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+class UUIDAndAgentCode(TestCase):
+    @classmethod
+    def setUp(self):
+        self.uuid_code = generate_uuid_and_agent_code()[0]
+        self.agent_code = generate_uuid_and_agent_code()[1]
+
+        return super().setUp(self)
+
+    def test_uuid_valid(self):
+        assert len(str(self.uuid_code)) == 36
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: All these can be done with assertEqual

File: itez/beneficiary/tests/test_utils.py (hunk 3 above)
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: This is a boolean and test that the value is uppercase not writing exact examples I gave
I was giving you just direction Cephas

File: itez/beneficiary/tests/test_utils.py (hunk 4 above)
Comment (sonlinux) [reply itez/beneficiary/tests/test_utils.py]: I want proper Docstrings for the class and each function
And ensure the method names are clear and saying what they are testing for exactly

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/tests/test_utils.py (hunk 3 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] add assertEqual

File: itez/beneficiary/tests/test_utils.py (hunk 5 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] assertEqual

File: itez/beneficiary/tests/test_utils.py (hunk 3 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x]  add assertEqual that checks if uppercase = true

File: itez/beneficiary/tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+from django.test import TestCase
+
+from itez.beneficiary.utils import generate_uuid_and_agent_code
+
+
+class UUIDAndAgentCode(TestCase):
+    @classmethod
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: This can not and is not a classmethod

File: itez/beneficiary/tests/test_utils.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    def setUp(self):
+        """
+         This method is called for every test that runs and helps with variable declaration
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: Break the Docstrings into 2 lines it's too long

File: itez/beneficiary/tests/test_utils.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        """
+        self.uuid_code = generate_uuid_and_agent_code()[0]
+        self.agent_code = generate_uuid_and_agent_code()[1]
+
+        return super().setUp(self)
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: This is just a set up why are we returning this
Are you referencing the Django docs on tests or you are just writing them

File: itez/beneficiary/tests/test_utils.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+    def tearDown(self):
+        """
+         This method is called for every test as well
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: This is a method not function

File: itez/beneficiary/tests/test_utils.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        """
+        return super().tearDown()
+
+    def test_uuid_valid(self):
+        """
+         This function tests for the validity of generated uuid.
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: Same here

File: itez/beneficiary/tests/test_utils.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        """
+        self.assertEqual(len(str(self.uuid_code)), 36)
+
+    def test_agent_code_length(self):
+        """
+         This function tests for length of agent code.
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: Same here

File: itez/beneficiary/tests/test_utils.py [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        """
+        self.assertEqual(len(str(self.agent_code)), 7)
+
+    def test_agent_code_upper(self):
+        """
+         This function tests for capitalization of agent code.
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: Same here

File: itez/beneficiary/tests/test_utils.py [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        """
+        self.assertEqual((self.agent_code), self.agent_code.upper())
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: We have . is uppercase method on strings use that here for a bool check

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/tests/test_utils.py (hunk 7 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] Docstrings breakdown

File: itez/beneficiary/tests/test_utils.py (hunk 9 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] change "function," to , "method."

File: itez/beneficiary/tests/test_utils.py (hunk 11 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] change, "function," to, "method."

File: itez/beneficiary/tests/test_utils.py (hunk 13 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] bool check implemented

File: itez/beneficiary/tests/test_utils.py (hunk 10 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] change, "function," to, "method."

File: itez/beneficiary/tests/test_utils.py (hunk 8 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] remove return statements

File: itez/beneficiary/tests/test_utils.py (hunk 6 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] remove @classmethod

File: itez/beneficiary/tests/test_utils.py (hunk 4 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] add proper Docstrings

File: itez/beneficiary/tests/test_utils.py (hunk 1 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] arrange imports properly

File: itez/beneficiary/tests/test_utils.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+from itez.beneficiary.utils import generate_uuid_and_agent_code
+
+
+class UUIDAndAgentCode(TestCase):
+
+    def setUp(self):
+        """
+         This method is called for every test that runs and 
+         helps with variable declaration
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: full stopat end of  line

File: itez/beneficiary/tests/test_utils.py [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+        """
+        self.uuid_code = generate_uuid_and_agent_code()[0]
+        self.agent_code = generate_uuid_and_agent_code()[1]
+
+    def tearDown(self):
+        """
+         This method is called for every test as well
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py]: full stop at end of line

File: itez/beneficiary/tests/test_utils.py [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+    def test_uuid_valid(self):
+        """
+         This method tests for the validity of generated uuid.
+        """
+        self.assertEqual(len(str(self.uuid_code)), 36)
+
+    def test_agent_code_length(self):
+        """
+         This method tests for length of agent code.
```
Comment (sonlinux) [inline itez/beneficiary/tests/test_utils.py:24]: full stop at end of line

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: itez/beneficiary/tests/test_utils.py (hunk 14 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] add full stop

File: itez/beneficiary/tests/test_utils.py (hunk 15 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py]: - [x] add full stop

File: itez/beneficiary/tests/test_utils.py (hunk 16 above)
Comment (cazterk) [reply itez/beneficiary/tests/test_utils.py:24]: - [x] add full stop + screenshot( ' passing tests ' )

working test

**Review by sonlinux — APPROVED** [review]
LGTM

**Comment by cazterk** [comment]
- [x] merge changes

**Pull request merged**